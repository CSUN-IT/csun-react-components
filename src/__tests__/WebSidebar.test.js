import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import WebSidebar from '../../src/components/web/Sidebar';

configure({ adapter: new Adapter() });

describe('WebSidebar', () => {
  it('Should be defined', () => {
    expect(WebSidebar).toBeDefined();
  });

  it('Renders component correctly', () => {
    const tree = shallow(
      <WebSidebar>
        <h3> Test Header </h3>
        <h4> Test Title </h4>
        <p> Example paragraph for WebSidebar component. </p>
      </WebSidebar>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Renders children correctly', () => {
    const tree = mount(
      <div>
        <WebSidebar>
          <h3> Test Header </h3>
          <h3> Test Title </h3>
        </WebSidebar>
      </div>,
    );
    expect(tree.find('h3')).toHaveLength(2);
  });
});
