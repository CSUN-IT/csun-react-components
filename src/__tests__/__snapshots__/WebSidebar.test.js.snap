// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebSidebar Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WebSidebar>
    <h3>
       Test Header 
    </h3>
    <h4>
       Test Title 
    </h4>
    <p>
       Example paragraph for WebSidebar component. 
    </p>
  </WebSidebar>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <h3>
             Test Header 
          </h3>,
          <h4>
             Test Title 
          </h4>,
          <p>
             Example paragraph for WebSidebar component. 
          </p>,
        ],
        " ",
      ],
      "className": "sidebar-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Test Header ",
        },
        "ref": null,
        "rendered": " Test Header ",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Test Title ",
        },
        "ref": null,
        "rendered": " Test Title ",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Example paragraph for WebSidebar component. ",
        },
        "ref": null,
        "rendered": " Example paragraph for WebSidebar component. ",
        "type": "p",
      },
      " ",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <h3>
               Test Header 
            </h3>,
            <h4>
               Test Title 
            </h4>,
            <p>
               Example paragraph for WebSidebar component. 
            </p>,
          ],
          " ",
        ],
        "className": "sidebar-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Test Header ",
          },
          "ref": null,
          "rendered": " Test Header ",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Test Title ",
          },
          "ref": null,
          "rendered": " Test Title ",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Example paragraph for WebSidebar component. ",
          },
          "ref": null,
          "rendered": " Example paragraph for WebSidebar component. ",
          "type": "p",
        },
        " ",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
